src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Text.Json.Serialization.Converters;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    /// <summary>
+    /// Creates and initializes serialization metadata for a type.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal sealed class SourceGenJsonTypeInfo<T> : JsonTypeInfo<T>
```

Comment:
I'm guessing this is just renaming `JsonTypeInfoInternal<T>` right? Any other changes the diff is not able to reveal here?

Reply:
it's rename + moving some source gen specific code to SourceGenJsonTypeInfoOfT.cs but mostly without any changes (I can't remember exactly but I tried to not do any redundant changes). The only possible changes are related to initializing properties and parameters which are now done as part of Configure but nothing major on the source gen part